### ISSUE TITLE
warn the user when attempting to screenshot blackness

### ISSUE THREAD

**Issue by derekriemer** [nvaccess/nvda#10576]
### Is your feature request related to a problem? Please describe.

Blind users often take screenshots for sighted colleagues. It is common, when using a privacy feature like NVDA's built in screen hiding feature, to screenshot while the screen curtain is pulled down. This causes the user to think they have taken a screenshot of some application, and instead they have taken a screenshot of blackness. They then send this black screenshot to a colleague asking for some sort of info, or making a claim about that screenshot, only to confuse the colleague, or provoke an embarrassing situation.

### Describe the solution you'd like

When the print screen key is pressed with or without modifiers, if screen curtain is enabled, intercept the user's attempt to take a screenshot and warn them that their screen is black. I propose the following dialog.

```
---------------------------------
Warning!
-------------------------
The screen is currently black, because screen curtain is enabled in NVDA. You are attempting to take a screenshot of a blank screen. How would you like to continue?
--------------------------------------------------------------------------------
Options grouping:
        [] brighten screen for 3 seconds and take a screenshot 
        [] disable screen curtain and do not take a screenshot
--------------------------------------------------------------------------------
          ok     |     cancel
```

### Describe alternatives you've considered

We could simply speak and braille a message. Or, we could disable the screen curtain then screenshot and silently re enable for the user.

**Comment by derekriemer** [comment]
Another dialog would just have 3 buttons, but I don't like long text in buttons which is why I made a radio group.

**Comment by DrSooom** [comment]
Too much text and not really practicable if you are using [Snagit by Tech Smith](https://www.techsmith.com/screen-capture.html) instead of Microsoft's Snipping Tool.

Therefore I suggest to hold the PrtSc during screen curtain is on, like it is currently the case with the NVDA modifier keys.
+ Screen curtain is enabled:
  + Speech/Braille message on the first PrtSc press: Press it twice/Double Press to disable the screen curtain and to start capturing a screenshot.
  + Second PrtSc press: Disable screen curtain and send the PrtSc key to the system.
+ Screen curtain is disabled:
  + Send the PrtSc key to the system; no interruptions or changes on NVDA's side.

Note: I'm not sure, which application wins on pressing PrtSc – maybe NVDA, maybe Snagit. This has to be tested.

**Comment by leonardder** [comment]
Implementation detail, if the ScreenCurtain inherrits from baseObject.ScriptableObject, you can provide a script on the curtain that captures print screen and disable itself for a short time before executing the gesture. In that case, I think a single flash of the screen is sufficient, a dialog is overkill.

**Comment by christopherpross** [comment]
I had implemented this, but after sending the keypress to the os, the curtain is reactivated to fast. So on the photo is blackness. A sleep will stop nvda running right? Is there a way to wait for 2 seconds and then call a method?
I also think of do simelar with the ocr functinality.
What do you think?

**Comment by CyrilleB79** [comment]
Use the following to call yourFunction with arguments arg1 and arg2:

```
import core
core.callLater(2000, yourFunction, arg1, arg2)
```

**Comment by ivnc** [comment]
Hi.

This solution, or a similar one, could be implemented also when performing OCR with screen curtain enabled.
Many users have recently reported that they were trying to execute Win10 OCR, without success due to screen curtain.

Regards.
